tests/phpunit/tests/comment/query.php

```diff
@@ -4941,4 +4941,53 @@ public static function filter_comments_pre_query_and_set_comments( $comments, $q
 
 		return array( get_comment( $c ) );
 	}
+
+	/**
+	 * @ticket 55460
+	 */
+	public function test_comment_cache_key_should_ignore_unset_params() {
+		global $wpdb;
+
+		$p = self::factory()->post->create();
+		$c = self::factory()->comment->create( array( 'comment_post_ID' => $p ) );
+
+		$_args = array(
+			'post_id'                   => $p,
+			'fields'                    => 'ids',
+			'update_comment_meta_cache' => true,
+			'update_comment_post_cache' => false,
+		);
+
+		$q1 = new WP_Comment_Query();
+		$q1->query( $_args );
+
+		$num_queries_all_args = $wpdb->num_queries;
+
+		// Unset 'fields'.
+		unset( $_args['fields'] );
+		$q2 = new WP_Comment_Query();
+		$q2->query( $_args );
+
+		$num_queries_skip_fields = $wpdb->num_queries;
```

Comment:
you can use get_num_queries() instead, and avoid global $wpdb

Reply:
That's a good alternative. I followed code from other tests.